clients/pkg/logentry/stages/logfmt_test.go

```diff
@@ -0,0 +1,312 @@
+package stages
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	util_log "github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+var testLogfmtYamlSingleStageWithoutSource = `
+pipeline_stages:
+- logfmt:
+    mapping:
+      out:  message
+      app:
+      duration:
+      unknown:
```

Comment:
q: should raise warning if any mapping not found in the parsed `logfmt` line?. Say in this case `unknown`?

Personally, faced these kind of issues if there is some typo in my mapping key.

(ignore me, if we already do that)